### PR TITLE
Bring GH pages parity to local config (Jekyll 3)

### PR DESCRIPTION
**Pull request by bf4**
Follows https://github.com/json-api/json-api/pull/988

Fixes needed for jekyll 3
- https://github.com/blog/2100-github-pages-now-faster-and-simpler-with-jekyll-3-0
- https://jekyllrb.com/docs/upgrading/2-to-3/

re: parity, see:
- https://github.com/github/pages-gem/pull/209/files
  `gem 'github-pages', group: :jekyll_plugins`
  - https://help.github.com/articles/configuring-jekyll/ via
    https://help.github.com/articles/using-jekyll-as-a-static-site-generator-with-github-pages/

### CODE REVIEW

**Comment by dgeb** [comment]
Thanks @bf4. 

Looks like we still need to remove `markdown: redcarpet` from our `_config.yml` to finish the transition.

**Comment by bf4** [comment]
@dgeb When I removed that, the syntax highlighting was off :)

File: _config.yml [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
+kramdown:
+  input: GFM
+  hard_wrap: false
+  template: '' # cannot customize
+  math_engine: mathjax # cannot customize
+  syntax_highligher: rouge # cannot customize
+jailed: false
+gems:
+  - jekyll-redirect-from
+markdown: redcarpet
```
Comment (bf4) [inline _config.yml:38]: originally I removed the markdown/redcarpet declarations, but then syntax highlighting was off and perhaps there were other formatting errors

**Comment by dgeb** [comment]
@bf4 fair enough, then :) I guess we have some upgrade work yet TBD. And I think we need to finish before the end of the month. Thanks for chipping away here.

**Comment by bf4** [comment]
> Thanks for chipping away here.

Really just sharing what I did when I posted something to my blog a month ago and everything broke :)

**Pull request merged**